net-misc/dropbear/dropbear-2018.76-r1.ebuild

```diff
@@ -0,0 +1,110 @@
+# Copyright 1999-2018 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit pam savedconfig user
+
+DESCRIPTION="small SSH 2 client/server designed for small memory environments"
+HOMEPAGE="https://matt.ucc.asn.au/dropbear/dropbear.html"
+SRC_URI="https://matt.ucc.asn.au/dropbear/releases/${P}.tar.bz2
+	https://matt.ucc.asn.au/dropbear/testing/${P}.tar.bz2"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~ia64 ~m68k ~mips ~ppc ~ppc64 ~s390 ~sh ~sparc ~x86 ~amd64-linux ~x86-linux"
+IUSE="bsdpty minimal multicall pam +shadow static +syslog +utmp zlib"
+
+LIB_DEPEND="zlib? ( sys-libs/zlib[static-libs(+)] )
+	dev-libs/libtommath[static-libs(+)]"
+RDEPEND="!static? ( ${LIB_DEPEND//\[static-libs(+)]} )
+	pam? ( virtual/pam )"
+DEPEND="${RDEPEND}
+	static? ( ${LIB_DEPEND} )"
+RDEPEND+=" pam? ( >=sys-auth/pambase-20080219.1 )"
+
+REQUIRED_USE="pam? ( !static )"
+
+set_options() {
+	progs=(
+		dropbear dbclient dropbearkey
+		$(usex minimal "" "dropbearconvert scp")
+	)
+	makeopts=(
+		MULTI=$(usex multicall 1 0)
+		STATIC=$(usex static 1 0)
+	)
+}
+
+src_prepare() {
+	eapply "${FILESDIR}/${PN}-$(ver_cut 1)-dbscp.patch"
+	sed \
+		-e '/SFTPSERVER_PATH/s:".*":"/usr/lib/misc/sftp-server":' \
+		default_options.h > localoptions.h || die
+	sed -i \
+		-e '/pam_start/s:sshd:dropbear:' \
+		svr-authpam.c || die
+	restore_config localoptions.h
+	default
+}
+
+src_configure() {
+	# XXX: Need to add libtomcrypt to the tree and re-enable this.
+	#	--disable-bundled-libtom
+	# We disable the hardening flags as our compiler already enables them
+	# by default as is appropriate for the target.
+
+	confopts=(
+		--disable-harden
+		$(use_enable zlib)
+		$(use_enable pam)
+		$(use_enable !bsdpty openpty)
+		$(use_enable shadow)
+		$(use_enable syslog)
+	)
+
+	if ! use utmp ; then
+		confopts+=(--disable-{lastlog,utmp{,x},wtmp{,x},loginfunc,putut{,x}line})
+	fi
+
+	econf "${confopts[@]}"
+}
+
+src_compile() {
+	set_options
+	emake "${makeopts[@]}" PROGRAMS="${progs[*]}"
+}
+
+src_install() {
+	set_options
+	emake "${makeopts[@]}" PROGRAMS="${progs[*]}" DESTDIR="${ED}" install
+	doman *.8
+	newinitd "${FILESDIR}"/dropbear.init.d dropbear
+	newconfd "${FILESDIR}"/dropbear.conf.d dropbear
+	dodoc CHANGES README SMALL MULTI
+
+	# The multi install target does not install the links right.
+	if use multicall ; then
+		cd "${ED}"/usr/bin
+		local x
+		for x in "${progs[@]}" ; do
+			ln -sf dropbearmulti ${x} || die "ln -s dropbearmulti to ${x} failed"
+		done
+		rm -f dropbear
```

Comment:
Why `-f`? You should know when it's present and when not. Also missing `|| die`.